src/physics.cpp

```diff
@@ -275,130 +283,135 @@ void sample_photon_reaction(Particle& p)
+                break;
+              }
+            }
+
+            double phi = 2.0*PI*prn(p.current_seed());
+            Direction u;
+            u.x = mu;
+            u.y = std::sqrt(1.0 - mu*mu)*std::cos(phi);
+            u.z = std::sqrt(1.0 - mu*mu)*std::sin(phi);
+
+            // Create secondary electron
+            p.create_secondary(u, E_electron, Particle::Type::electron);
+
+            // Allow electrons to fill orbital and produce auger electrons
+            // and fluorescent photons
+            element.atomic_relaxation(shell, p);
+            p.event_ = TallyEvent::ABSORB;
+            p.event_mt_ = 533 + shell.index_subshell;
+            p.alive_ = false;
+            p.E_ = 0.0;
+            return;
           }
         }
+      }
+      prob = prob_after;
 
-        double phi = 2.0*PI*prn(p.current_seed());
-        Direction u;
-        u.x = mu;
-        u.y = std::sqrt(1.0 - mu*mu)*std::cos(phi);
-        u.z = std::sqrt(1.0 - mu*mu)*std::sin(phi);
+      // Pair production
+      prob += micro.pair_production;
+      if (prob > cutoff) {
+        double E_electron, E_positron;
+        double mu_electron, mu_positron;
+        element.pair_production(alpha, &E_electron, &E_positron,
+          &mu_electron, &mu_positron, p.current_seed());
 
         // Create secondary electron
+        Direction u = rotate_angle(p.u(), mu_electron, nullptr, p.current_seed());
         p.create_secondary(u, E_electron, Particle::Type::electron);
 
-        // Allow electrons to fill orbital and produce auger electrons
-        // and fluorescent photons
-        element.atomic_relaxation(shell, p);
+        // Create secondary positron
+        u = rotate_angle(p.u(), mu_positron, nullptr, p.current_seed());
+        p.create_secondary(u, E_positron, Particle::Type::positron);
+
         p.event_ = TallyEvent::ABSORB;
-        p.event_mt_ = 533 + shell.index_subshell;
+        p.event_mt_ = PAIR_PROD;
         p.alive_ = false;
         p.E_ = 0.0;
-        return;
       }
-    }
-  }
-  prob = prob_after;
-
-  // Pair production
-  prob += micro.pair_production;
-  if (prob > cutoff) {
-    double E_electron, E_positron;
-    double mu_electron, mu_positron;
-    element.pair_production(alpha, &E_electron, &E_positron,
-      &mu_electron, &mu_positron, p.current_seed());
-
-    // Create secondary electron
-    Direction u = rotate_angle(p.u(), mu_electron, nullptr, p.current_seed());
-    p.create_secondary(u, E_electron, Particle::Type::electron);
-
-    // Create secondary positron
-    u = rotate_angle(p.u(), mu_positron, nullptr, p.current_seed());
-    p.create_secondary(u, E_positron, Particle::Type::positron);
-
-    p.event_ = TallyEvent::ABSORB;
-    p.event_mt_ = PAIR_PROD;
-    p.alive_ = false;
-    p.E_ = 0.0;
-  }
+    } // add by Yuan
+  } // add by Yuan
+	
+  // weight window  add by Yuan
+  // Check if weight window is used
+  if (settings::weightwindow && settings::p_ww && p.E_ > 0.0) split_particle(p);
```

Comment:
Can we do the energy check inside of split_particle  and return immediately to make this if statement less verbose?

Reply:
Yeah, I could move the energy check into the aplit_particle function.